smallrye-reactive-messaging-provider/src/test/java/io/smallrye/reactive/messaging/inject/EmitterInjectionTest.java

```diff
@@ -137,14 +141,16 @@ public void testWithNull() {
         assertThat(bean.hasCaughtNullMessage()).isTrue();
     }
 
-    @Test(expected = DeploymentException.class)
+    @Test(expected = IllegalStateException.class)
     public void testWithMissingStream() {
-        installInitializeAndGet(BeanWithMissingStream.class);
+        // The error is only thrown when a message is emitted as the subscription can be delayed.
+        installInitializeAndGet(BeanWithMissingStream.class).emitter().send(Message.of("foo"));
     }
 
-    @Test(expected = DeploymentException.class)
+    @Test(expected = IllegalStateException.class)
```

Comment:
I see what you meant. This will impact spec and tck change because the validation is at runtime instead of startup time.

Reply:
Yes, That's what I meant. This is actually required for https://github.com/eclipse/microprofile-reactive-messaging/issues/3